google-cloud-logging/src/main/java/com/google/cloud/logging/GaeFlexLoggingEnhancer.java

```diff
@@ -65,13 +70,29 @@ public GaeFlexLoggingEnhancer() {
 
   @Override
   public void enhanceMonitoredResource(Builder builder) {
-    if (gaeInstanceId != null) {
-      if (System.getenv("GAE_SERVICE") != null) {
-        builder.addLabel("module_id", System.getenv("GAE_SERVICE"));
-      }
-      if (System.getenv("GAE_VERSION") != null) {
-        builder.addLabel("version_id", System.getenv("GAE_VERSION"));
+    if (gaeInstanceId == null) {
+      return;
+    }
+    if (System.getenv("GAE_SERVICE") != null) {
+      builder.addLabel("module_id", System.getenv("GAE_SERVICE"));
+    }
+    if (System.getenv("GAE_VERSION") != null) {
+      builder.addLabel("version_id", System.getenv("GAE_VERSION"));
+    }
+    try {
+      HttpURLConnection metadata = (HttpURLConnection) new URL(
+          "http://metadata.google.internal/computeMetadata/v1/instance/zone").openConnection();
+      metadata.setRequestProperty("Metadata-Flavor", "Google");
```

Comment:
Maybe just add a TODO at the other location - I don't want to perturb anything more than necessary right now.